File: python/1-1.py
# 2分法による方程式の解法

EPS = 0.0001        # 許容誤差

def nibun( a, b ):
    while True:
        c = ( a + b ) / 2.0     # 2分計算
        if( ( func_y(c) * func_y(a) ) < 0 ): b = c  # 式(1.2)
        else: a = c                                 # 式(1.3)
        if not( abs( a - b ) > EPS ): break         # 収束判定式(1.4)の変形
    return c

# 関数 y = x^3 + x - 1
def func_y( x ):
    return pow( x, 3.0 ) + x - 1.0

# 初期値設定
a = 0.0
b = 1.0

print( "x^3 + x -1 = 0 の2分法による数値計算")
print( f"初期値a = {a:6.3f}")
print( f"初期値b = {b:6.3f}")
x = nibun( a, b )
print( f"近似解x = {x:6.3f}")       # 解打出し

File: julia/1-1.jl
# 2分法による方程式の解法

using Printf

const ε = 0.0001        # 許容誤差

function nibun( a, b )
    c = 0
    while true
        c = ( a + b ) / 2.0     # 2分計算
        if ( func_y(c) * func_y(a) ) < 0    # 式(1.2)
            b = c
        else
            a = c       # 式(1.3)
        end
        if !( abs( a - b) > ε )     # 収束判定式(1.4)の変形
            break
        end
    end
    return c
end

func_y( x ) = x^3.0 + x - 1.0

# 初期値設定
a = 0.0
b = 1.0

println( "x^3 + x - 1 = 0 の2分法による数値計算\n" )
@printf( "初期値a = %6.3lf\n", a )
@printf( "初期値b = %6.3lf\n", b )
x = nibun( a, b )
@printf( "近似解x = %6.3lf\n", x )      # 解打出し
